### PR TITLE
Expense date added to expense dialog

### LINKED ISSUE #94: Add date input in dialog
**Current behavior:** expense dates are currently set to whenever the expense is added
**Expected behavior:** let the user choose the expense date and correctly insert it to the expense table

### PR DESCRIPTION
**Pull request by VictoriaKGN**
users were not able to pick a date for expenses, which is now added.
formik did not like date pickers so had no choice but to go with a formatted text.

image
image

closes #94

### CODE REVIEW

File: client/src/components/ExpenseDialog/ExpenseDialog.tsx [hunk 1]
```diff
@@ -86,100 +127,116 @@ const ExpenseDialog: React.FC<ExpenseDialogProps> = ({
...
+                },
+              }}
+              InputLabelProps={{
+                style: { color: theme.palette.primary.contrastText },
+              }}
+              style={{ width: 300 }}
+            />
+            <TextField
+              id="createdAt"
+              label="Date"
```
Comment (BarbzCodez) [inline client/src/components/ExpenseDialog/ExpenseDialog.tsx:156]: I think it wouldn't hurt to add the format in the label just to make it super explicit like `Date (yyyy/mm/dd)`

Comment (VictoriaKGN) [reply client/src/components/ExpenseDialog/ExpenseDialog.tsx:156]: wouldn't that be redundant? users wont be able to enter a date in a different format, and all rows in the table will have that one format

Comment (BarbzCodez) [reply client/src/components/ExpenseDialog/ExpenseDialog.tsx:156]: Didnt realize it gives an error when it has the incorrect date format so tis all gud
image

**Review by BarbzCodez — APPROVED** [review]
image

**Review by dlarocque — APPROVED** [review]

**Pull request merged**